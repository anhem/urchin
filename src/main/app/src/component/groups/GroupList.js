import React, {Component} from "react";

class GroupList extends Component {

    render() {
        let groups = this.props.groups || [];
        let groupItems = groups.map((item, index) =>
            <li key={index.toString()}>
                <a href={`/groups/${item.groupId}`}>{item.groupName}</a>
            </li>
        );
        return (
            <div>
                <a href="/groups/new">new group</a>
                <h2>grouplist</h2>
                <ul>
                    {groupItems}
                </ul>
            </div>
        )
    }
}

export default GroupList