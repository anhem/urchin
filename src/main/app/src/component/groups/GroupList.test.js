import React from 'react';
import {mount} from 'enzyme'
import GroupList from "./GroupList";

describe('GroupList', () => {

    it('renders without crashing', () => {
        expect(mount(<GroupList/>).length).toEqual(1);
    });
});
