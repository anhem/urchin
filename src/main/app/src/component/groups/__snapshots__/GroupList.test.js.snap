// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupList match snapshot 1`] = `
<div>
  <a
    href="/groups/new"
  >
    new group
  </a>
  <h2>
    grouplist
  </h2>
  <ul />
</div>
`;
